Mathlib/CategoryTheory/Monoidal/NaturalTransformation.lean

```diff
@@ -0,0 +1,260 @@
+    tensor := fun X Y => by
+      dsimp
+      rw [Iso.comp_inv_eq, assoc, tensor', ← tensor_comp_assoc,
+        Iso.inv_hom_id, Iso.inv_hom_id, tensor_id, id_comp] }
+  hom_inv_id := by
+    apply MonoidalNatTrans.ext
+    ext x
+    exact (app x).hom_inv_id
+  inv_hom_id := by
+    apply MonoidalNatTrans.ext
+    ext x
+    exact (app x).inv_hom_id
+#align category_theory.monoidal_nat_iso.of_components CategoryTheory.MonoidalNatIso.ofComponents
+
+@[simp]
+theorem ofComponents.hom_app (app : ∀ X : C, F.obj X ≅ G.obj X) (naturality) (unit) (tensor) (X) :
+    (ofComponents app naturality unit tensor).hom.app X = (app X).hom :=
+  rfl
+#align category_theory.monoidal_nat_iso.of_components.hom_app CategoryTheory.MonoidalNatIso.ofComponents.hom_app
+
+@[simp]
+theorem ofComponents.inv_app (app : ∀ X : C, F.obj X ≅ G.obj X) (naturality) (unit) (tensor) (X) :
+    (ofComponents app naturality unit tensor).inv.app X = (app X).inv := by simp [ofComponents]
+#align category_theory.monoidal_nat_iso.of_components.inv_app CategoryTheory.MonoidalNatIso.ofComponents.inv_app
+
+instance isIso_of_isIso_app (α : F ⟶ G) [∀ X : C, IsIso (α.app X)] : IsIso α :=
+  ⟨(IsIso.of_iso
+        (ofComponents (fun X => asIso (α.app X)) (fun f => α.toNatTrans.naturality f) α.unit
+          α.tensor)).1⟩
+#align category_theory.monoidal_nat_iso.is_iso_of_is_iso_app CategoryTheory.MonoidalNatIso.isIso_of_isIso_app
+
+end MonoidalNatIso
+
+noncomputable section
+
+/-- The unit of a monoidal equivalence can be upgraded to a monoidal natural transformation. -/
+@[simp] -- Porting note: changed to simp
+def monoidalUnit (F : MonoidalFunctor C D) [IsEquivalence F.toFunctor] :
+    LaxMonoidalFunctor.id C ⟶ F.toLaxMonoidalFunctor ⊗⋙ (monoidalInverse F).toLaxMonoidalFunctor :=
+  let e := F.toFunctor.asEquivalence
+  { toNatTrans := e.unit
+    tensor := fun X Y => by
+      -- This proof is not pretty; golfing welcome!
+      dsimp
+      simp only [Adjunction.homEquiv_unit, Adjunction.homEquiv_naturality_right,
+        id_comp, assoc]
+      simp only [← Functor.map_comp, assoc]
+      erw [e.counit_app_functor, e.counit_app_functor,
+        F.toLaxMonoidalFunctor.μ_natural, IsIso.inv_hom_id_assoc]
+      simp only [CategoryTheory.IsEquivalence.inv_fun_map]
+      slice_rhs 2 3 => erw [Iso.hom_inv_id_app]
+      dsimp
+      simp only [CategoryTheory.Category.id_comp]
+      slice_rhs 1 2 =>
+        rw [← tensor_comp, Iso.hom_inv_id_app, Iso.hom_inv_id_app]
+        dsimp
+        rw [tensor_id]
+      simp }
+#align category_theory.monoidal_unit CategoryTheory.monoidalUnit
+
+instance (F : MonoidalFunctor C D) [IsEquivalence F.toFunctor] : IsIso (monoidalUnit F) :=
+  haveI : ∀ X : C, IsIso ((monoidalUnit F).toNatTrans.app X) := by
+    intros
+    dsimp
+    infer_instance
+  MonoidalNatIso.isIso_of_isIso_app _
+
+/-- The counit of a monoidal equivalence can be upgraded to a monoidal natural transformation. -/
+@[simp] -- Porting note: changed to simp
```

Comment:
I don't think this is the right thing to do here. `simp` does not do the same thing as `simps`. I think it is the `let` that is stopping `simps` from working here, so either `simps` should be changed to work with `let`s or the definition should be rewritten without `let` and with a porting note explaining the change

Reply:
Hmmm.... it seems that the `let` was not the issue. Something else is going on. Will investigate soon.

Reply:
Both `[simp]` have been reverted to `simps!` and the errors fixed by specifying the toNatTrans projection.